### PR TITLE
Remove extraneous files, remove user stories

### PR DESCRIPTION
**Pull request by nhcarrigan**
Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

Related https://github.com/freeCodeCamp/freeCodeCamp/issues/39721

Should not be merged until https://github.com/freeCodeCamp/freeCodeCamp/pull/40219 hits production

* Remove user stories from HTML
* Standardise README
* Remove unneeded files

### CODE REVIEW

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@nhcarrigan As I stated in https://github.com/freeCodeCamp/freeCodeCamp/pull/40219, I do not see any `test('test text', function(done) { })`, so it is unclear how many actual tests need to be written.

**Comment by nhcarrigan** [comment]
Sure thing - I will push an update later this afternoon 🙂

**Comment by RandellDawson** [comment]
@nhcarrigan Would also like to see the following removed from `/routes/api.js`

```js
/*
*
*
*       Complete the API routing below
*
*
*/
```

and the following from the `/test/2_functional-tests.js` file:
```js
/*
*
*
*       FILL IN EACH FUNCTIONAL TEST BELOW COMPLETELY
*       -----[Keep the tests in the same order!]-----
*       (if additional are added, keep them at the very end!)
*/
```
If you go with something like was done boilerplate in  be changes to explain what test to write like is done in boilerplate PR https://github.com/freeCodeCamp/boilerplate-mochachai/pull/27, then you can modify the `/learn` page like https://github.com/freeCodeCamp/freeCodeCamp/pull/39976 referencing the test by a commented number (i.e. `// #1`).

**Comment by nhcarrigan** [comment]
> If you go with something like was done boilerplate in be changes to explain what test to write like is done in boilerplate PR freeCodeCamp/boilerplate-mochachai#27, then you can modify the /learn page like freeCodeCamp/freeCodeCamp#39976 referencing the test by a commented number (i.e. // #1).

This step might not be necessary. I'm fairly confident that the order the camper writes the tests in is ultimately irrelevant for these projects - we access the tests through a route defined in `fcctesting.js` and iterate through the returned array, for the client-side validation. As such, the tests the camper writes aren't individually verified - instead we check for length, a `passing` value on each test, and at least one `assert` on each test. 

If we want to strengthen that, and verify that the tests a camper writes are actual tests (and not a series of `assert.isTrue(true)`, for example), we'll need to adjust a good number of the /learn projects.

**Comment by RandellDawson** [comment]
I was under the impression that the order of the tests made a difference, which is why the comment existed to begin with.

**Comment by nhcarrigan** [comment]
```js
        const getTests = await $.get(getUserInput('url') + '/_api/get-tests' );
        assert.isArray(getTests);
        const functTests = getTests.filter((test) => {
          return !!test.context.match(/Functional Tests ->/ig);
        });
        assert.isAtLeast(functTests.length, 6, 'At least 6 tests passed');
        functTests.forEach(test => {
          assert.equal(test.state, 'passed', 'Tests in Passed State');
          assert.isAtLeast(test.assertions.length, 1, 'At least one assertion per test');
        });
```

As it stands now, the approach we've been using has been this.

File: tests/2_functional-tests.js
Comment (RandellDawson) [inline tests/2_functional-tests.js:13]: @nhcarrigan I think we may have a different issue.  We are depending on the user reading the text of the test to know what to write for this particular test. Ideally, the `/learn` page would explain what tests need to be written and not depend on the user to be able to read the text of the test in the boilerplate.  Why, because all instructions or instruction-like parts of a project need to be translated and we can not translate the boilerplate code.

Comment (nhcarrigan) [reply tests/2_functional-tests.js:13]: Ah, I see. 

Okay - I can run a refactor on this (and the other) to:
* Add number comments to identify the tests, as in the mocha chai boilerplate
* Move the corresponding instructions to a numeric list in the `/learn` instructions

I think this will need to be done for all of the QA and InfoSec projects, though, as they all follow this format.

Comment (RandellDawson) [reply tests/2_functional-tests.js:13]: > I think this will need to be done for all of the QA and InfoSec projects, though, as they all follow this format.

Yes, I agree.  @moT01 Maybe we have this done in a separate PR for each QA and InfoSec project, so we can at least merge these "as-is"?  We will just not close freeCodeCamp/freeCodeCamp#39721 until this has been done for all these projects? Plus the scope of those new PRs will be very specific.  Or, should we them go ahead and fix each of the existing open PRs?

Comment (moT01) [reply tests/2_functional-tests.js:13]: Tough to say, many of these have certainly gone outside the scope of the defined issue, but I'm fine with that since changing things later will result in another round of PR's and reviews for each boilerplate with the issue. We could look at them case by case - this particular PR and it's related learn PR came in yesterday and are just starting the review process, so I wouldn't mind changing these before they're merged. Others that have been reviewed thoroughly and are ready to be merged, it may not make sense to now change those up further. I suppose it also depends if @nhcarrigan wants to make the changes. If he's willing - which it seems he is - we could try and change them - I don't mind saving it for another issue/PR either.

To try and clarify, what would you do here?
```diff
-  test('create 2 new threads (one is deleted in the delete test)', function(done) {
+  test('#1', function(done) {
```
Number all the tests like that, and then put in the instructions on learn a list of the test text and the number they go with? I know we've talked about that a little bit, but maybe we should see what others think about it before we make a decision to change them all. If we really want to move all the instructions to learn, I don't see much else for options though. And if these don't make their way to learn somehow, they won't get translated. I do kind of feel like this could use a little discussion - maybe someone can come up with a better method or maybe people will think this is too confusing or something.

Comment (RandellDawson) [reply tests/2_functional-tests.js:13]: I like this approach.  However, I think the verbiage for the tests should be in the instructions sections above and the last user story in the tests would refer the user to fulfill the tests specified in the instructions.

Comment (nhcarrigan) [reply tests/2_functional-tests.js:13]: > I think the verbiage for the tests should be in the instructions sections above and the last user story in the tests would refer the user to fulfill the tests specified in the instructions.

This was my thought as well - I thought of something like:
```md
## Instructions
<section id='instructions'>

1. Set `NODE_ENV` to test without quotes when ready to write tests and DB to your databases connection string (in `.env`)
2. Recommended to create controllers/handlers and handle routing in `routes/api.js`
3. You will add any security features to `server.js`

Write the following unit tests in `tests/1_unit-tests.js`:
1. Test description

Write the following functional tests in `tests/2_functional-tests.js`:
1. Test description
</section>
```

That way we could remove the "unit test" section if applicable.

Comment (RandellDawson) [reply tests/2_functional-tests.js:13]: @moT01 and I were just discussing how if we go with putting the instructions for the tests to write in the `Instructions` section, then maybe we could remove all the `test('#1', function(done) {` and just leave the `suite`s to force the user to do more of the work.  The instructions would just tell them what tests to write and reference which `suite` to put them in.  What do you think?

Comment (nhcarrigan) [reply tests/2_functional-tests.js:13]:
```js
const chaiHttp = require('chai-http');
const chai = require('chai');
const assert = chai.assert;
const server = require('../server');

chai.use(chaiHttp);

suite('Functional Tests', function() {
    
});
```

So reducing it down to this? I like that approach. The client test will confirm the user has written enough tests to match the user story requirements.
(I'd be willing to go one step further and remove the imports as well)

Comment (RandellDawson) [reply tests/2_functional-tests.js:13]: > (I'd be willing to go one step further and remove the imports as well)

Which ones were you thinking?

Comment (nhcarrigan) [reply tests/2_functional-tests.js:13]: Bringing it down to only:
```js
suite('Functional Tests', function() {
    
});
```

It'll require more thinking from the user - though it could also create more confusion than we'd like.

Comment (RandellDawson) [reply tests/2_functional-tests.js]: I could be ok with this.  @moT01 @scissorsneedfoodtoo What do you think?

Comment (scissorsneedfoodtoo) [reply tests/2_functional-tests.js]: @RandellDawson, that looks good to me. By this point learners should be familiar with import statements. Simplifying the test files like this would also give them a chance to write tests in either BDD or TDD styles.

Comment (nhcarrigan) [reply tests/2_functional-tests.js]: Alright - I can start shipping this change to the other projects as well, then. 🙂

Comment (scissorsneedfoodtoo) [reply tests/2_functional-tests.js]: Awesome! Thank you for all of your hard work on these projects @nhcarrigan, and for everything else you're helping us with.

Comment (nhcarrigan) [reply tests/2_functional-tests.js]: Always glad to help! I'll start shipping these changes tomorrow!

File: package.json
Comment (moT01) [inline package.json:19]:
```suggestion
		"mongodb": "^3.5.8"
```

**Review by moT01 — COMMENTED** [review]
I'm pretty sure zombie isn't needed for this one? I would have just left it, but I was having issues with mongo. I tried to use the code from the demo project to pass this and the tests weren't passing from an old version of mongo. After I updated, they passed. I'm wondering if we should include a newer package? Or if we want to default users to an older version - I would say to update it since I was having this issue. We would probably want to update the package-lock file as well. Any thoughts? Note that `mongodb: '^3.5.8'` is the version the demo project is using.

File: package.json
Comment (nhcarrigan) [reply package.json:19]: Was there a decision or proposal on one of these PRs to remove the DB dependency entirely from the package.json?
Or am I mis-remembering?

Comment (RandellDawson) [reply package.json:19]: I believe it was just `mongoose`, but I could be wrong.

Comment (moT01) [reply package.json:19]: I think you're right on that one. I don't see anywhere the connection is defined in the boilerplate - so I am going to say to yea, just remove it. Maybe @RandellDawson can give a quick second thought.

Comment (moT01) [reply package.json:19]: Pretty sure I noticed it was gone from at least one of the other boilerplates. Seems like we should figure it out.

Comment (RandellDawson) [reply package.json:19]: @nhcarrigan Actually, you are correct.  I validated the mongo and zombie packages can be removed from this boilerplate (and probably the other boilerplate).  As long as the packages are not referenced by any other file in the boilerplates, they can be removed.  This allows the users to add whatever they want for a database.

Comment (nhcarrigan) [reply package.json]: Alright, those have been removed and a new `package-lock.json` has also been generated.

**Comment by RandellDawson** [comment]
If we are going to remove the database, then maybe, `sample.env` should remove the `DB=` to let the user create their own environment variable?

**Comment by nhcarrigan** [comment]
> If we are going to remove the database, then maybe, `sample.env` should remove the `DB=` to let the users create their own environment variable?

We will need to remove the comment from the `/learn` PR as well, then - as it does specify ` and DB to your databases connection string (in `.env`)`

**Comment by RandellDawson** [comment]
> > If we are going to remove the database, then maybe, `sample.env` should remove the `DB=` to let the users create their own environment variable?
> 
> We will need to remove the comment from the `/learn` PR as well, then - as it does specify `and DB to your databases connection string (in`.env`)`
> 
> I can certainly do that, though, if that's what we want

@moT01 What do you think?

**Comment by RandellDawson** [comment]
@nhcarrigan On second thought, just leave the `DB=` in the `sample.env` file. If you want to create an issue to discuss removing it from all challenges, that is probably a better route and then we can create a PR if needed.  That way, we can get this one merged sooner.

**Review by RandellDawson — APPROVED** [review]
LGTM.  I am good with these changes "as-is".  @moT01 If you approve, then we can merge this PR once the changes from https://github.com/freeCodeCamp/freeCodeCamp/pull/40219 hit production.

**Comment by Sky020** [comment]
@RandellDawson @moT01 The associated PR has hit production

**Comment by RandellDawson** [comment]
@moT01 If you are good with these changes, then merge.

**Pull request merged**
